internal/provider/models/processors/test/filter_test.go

```diff
@@ -0,0 +1,291 @@
+							expressions_group = [
+								{
+									expressions = [
+										{
+											field        = ".label"
+											operator     = "equal"
+											value_string = "account"
+										},
+										{
+											field        = ".app_name"
+											operator     = "ends_with"
+											value_string = "service"
+										},
+									]
+									logical_operation = "OR"
+								},
+								{
+									expressions_group = [
+										{
+											expressions = [
+												{
+													field = ".level"
+													operator = "greater_or_equal"
+													value_number = 300
+												},
+												{
+													field = ".tag"
+													operator = "contains"
+													value_string = "error"
+												}
+											]
+											logical_operation = "OR"
+										}
+									]
+								}
+							]
+						}
+					}`,
+				Check: resource.ComposeTestCheckFunc(
+					StateHasExpectedValues("mezmo_filter_processor.nested_expression", map[string]any{
+						"pipeline_id":                     "#mezmo_pipeline.test_parent.id",
+						"title":                           "processor title",
+						"description":                     "processor desc",
+						"generation_id":                   "0",
+						"inputs.#":                        "1",
+						"conditional.expressions_group.#": "2",
+						"conditional.expressions_group.0.expressions.#":                                  "2",
+						"conditional.expressions_group.0.logical_operation":                              "OR",
+						"conditional.expressions_group.0.expressions.0.field":                            ".label",
+						"conditional.expressions_group.0.expressions.0.operator":                         "equal",
+						"conditional.expressions_group.0.expressions.0.value_string":                     "account",
+						"conditional.expressions_group.0.expressions.1.field":                            ".app_name",
+						"conditional.expressions_group.0.expressions.1.operator":                         "ends_with",
+						"conditional.expressions_group.0.expressions.1.value_string":                     "service",
+						"conditional.expressions_group.1.expressions_group.0.expressions.#":              "2",
+						"conditional.expressions_group.1.expressions_group.0.logical_operation":          "OR",
+						"conditional.expressions_group.1.expressions_group.0.expressions.0.field":        ".level",
+						"conditional.expressions_group.1.expressions_group.0.expressions.0.operator":     "greater_or_equal",
+						"conditional.expressions_group.1.expressions_group.0.expressions.0.value_number": "300",
+						"conditional.expressions_group.1.expressions_group.0.expressions.1.field":        ".tag",
+						"conditional.expressions_group.1.expressions_group.0.expressions.1.operator":     "contains",
+						"conditional.expressions_group.1.expressions_group.0.expressions.1.value_string": "error",
+					}),
+				),
+			},
+			// Connect outputs to destinations
```

Comment:
Why do we have to test this?  Connecting nodes is normal vector stuff, not having to do with this processor.  Am I missing something?

Reply:
Twas mostly for my sanity.  I wondered for a moment if filter had multiple outputs